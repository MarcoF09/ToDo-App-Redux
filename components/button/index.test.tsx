import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import { Button } from './index'

describe('should render button', () => {
  it('renders correctly', () => {
    const component = shallow(<Button text="Dummy" onPress={jest.fn()} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  // it('should call onPress', () => {
  //   const mockOnPress = jest.fn();
  //   const wrapper = shallow(<Button text="Dummy" onPress={mockOnPress} />);
  //   const instance = wrapper.instance();

  //   instance.onPress();

  //   expect(mockOnPress).toHaveBeenCalled();
  //   expect(mockOnPress).toHaveBeenCalledTimes(1);
  // });
})
