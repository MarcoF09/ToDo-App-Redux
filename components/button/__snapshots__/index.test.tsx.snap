// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[MockFunction]}
    underlayColor="white"
  >
    <View>
      <Text
        style={
          Object {
            "color": "#ff197b",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 14,
            "lineHeight": 32,
          }
        }
      >
        Dummy
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;
